packages/run-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -229,23 +216,30 @@ export const makeVaultManager = (
       getAmountIn(quote),
     );
 
-    enqueueToLiquidate(
-      prioritizedVaults.entriesPrioritizedGTE(quoteRatioPlusMargin),
-    );
-
     outstandingQuote = undefined;
-    // Ensure all vaults complete
-    await executeLiquidation();
+
+    // Liquidate serially
+    for (const [key, vault] of prioritizedVaults.entriesPrioritizedGTE(
+      quoteRatioPlusMargin,
+    )) {
+      // eslint-disable-next-line no-await-in-loop -- We want each liquidation to happen in a separate turn
+      await liquidateAndRemove([key, vault]);
```

Comment:
This is actually exactly the scenario that the `await` rule is designed to prevent, so I don't think this is an allowable exception. If `entriesPrioritizedGTE` returns an empty iteration, then line 229 will execute in the same turn, whereas if it does not, it would execute in a different turn.